[webkitbugspy] Adopt new radar authentication methods

JonWBedard opened the pull request:
#### 0d3dd05d419f2f551ce3d52dc8e90177ea70cd38
<pre>
[webkitbugspy] Adopt new radar authentication methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=293388">https://bugs.webkit.org/show_bug.cgi?id=293388</a>
<a href="https://rdar.apple.com/151807028">rdar://151807028</a>

Reviewed by Brianna Fan.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py: Import timezone
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
(RadarModel.commit_changes): Set addedBy if unset.
(RadarClient.current_user): Add mock function.
(RadarClient.create_radar):
(Radar.AuthenticationStrategyNarrative): Added.
(Radar.AuthenticationStrategySystemAccountOAuth): Added.
(Radar.AuthenticationStrategyAppleConnect): Added.
(Radar.DiagnosisEntry.__init__):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.authentication): Leverage new authentication methods.
(Tracker.me): Use current_user to determine &quot;me&quot;.

Canonical link: <a href="https://commits.webkit.org/295608@main">https://commits.webkit.org/295608@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e30f115970921a53ad2f0343afb1d9eb6d4bd13c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104530 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24240 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14658 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109738 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55201 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106570 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24624 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32784 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79362 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107536 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19152 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94342 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59688 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/104008 "Passed tests") | [💥 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18928 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12415 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54572 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88634 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12471 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112124 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31691 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23334 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88407 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32055 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90575 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88076 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32979 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10749 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/27008 "Hash e30f1159 for PR 45726 does not build (failure)") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17130 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31618 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31410 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34749 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32970 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
```diff
@@ -218,6 +219,11 @@ def keywords(self):
     def commit_changes(self):
         self.client.parent.request_count += 1
 
+        # Anything wihtout someone who added it was made by the current user
```
`wihtout` -> `without`

JonWBedard committed:
[webkitbugspy] Adopt new radar authentication methods
https://bugs.webkit.org/show_bug.cgi?id=293388
rdar://151807028

Reviewed by Brianna Fan.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py: Import timezone
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
(RadarModel.commit_changes): Set addedBy if unset.
(RadarClient.current_user): Add mock function.
(RadarClient.create_radar):
(Radar.AuthenticationStrategyNarrative): Added.
(Radar.AuthenticationStrategySystemAccountOAuth): Added.
(Radar.AuthenticationStrategyAppleConnect): Added.
(Radar.DiagnosisEntry.__init__):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.authentication): Leverage new authentication methods.
(Tracker.me): Use current_user to determine "me".

Canonical link: https://commits.webkit.org/295608@main
